package hector.resources;

import "Scheme.proto";

option optimize_for = SPEED;

message WebResource {
	optional string url = 1;		// complete URL
	optional fixed32 ip4_addr = 2;		// IPv4 address
	optional fixed64 ip6_addr_1 = 3;	// IPv6 address, part 1
	optional fixed64 ip6_addr_2 = 4;	// IPv6 address, part 2
	repeated string header_names = 5;	// response header field names
	repeated string header_values = 6;	// response header field values
	optional int32 redirect_count = 7;	// number of redirects
	optional bytes content = 8;		// content of a resource
	optional fixed32 scheduled = 9;		// last touched this WR
}
